src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -314,17 +322,18 @@ internal void OnSolutionBuildCompleted()
                 {
                     await _postBuildAndErrorListRefreshTaskQueue.LastScheduledTask.ConfigureAwait(false);
                 }
-            }, cancellationToken);
+            }, GetApplicableCancellationToken(inProgressState));
         }
 
         /// <summary>
         /// Core method that de-dupes live and build diagnostics at the completion of build.
         /// It raises diagnostic update events for both the Build-only diagnostics and Build + Intellisense diagnostics
         /// in the error list.
         /// </summary>
-        private async Task SyncBuildErrorsAndReportOnBuildCompletedAsync(DiagnosticAnalyzerService diagnosticService, InProgressState inProgressState, CancellationToken cancellationToken)
+        private async Task SyncBuildErrorsAndReportOnBuildCompletedAsync(DiagnosticAnalyzerService diagnosticService, InProgressState inProgressState)
         {
             var solution = inProgressState.Solution;
+            var cancellationToken = inProgressState.CancellationToken;
             var (allLiveErrors, pendingLiveErrorsToSync) = inProgressState.GetLiveErrors(cancellationToken);
```

Comment:
It seems simpler to just remove the `cancellationToken` parameter

Reply:
I agree! Missed this one but will fix it up before merge.